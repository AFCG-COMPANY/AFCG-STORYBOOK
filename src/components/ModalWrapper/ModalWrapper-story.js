import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import ModalWrapper from '../ModalWrapper';
import TextInput from '../TextInput';
import Select from '../Select';
import SelectItem from '../SelectItem';
import RadioButtonGroup from '../RadioButtonGroup';
import RadioButton from '../RadioButton';

const modalProps = {
  onBlur: action('onBlur'),
  onClick: action('onClick'),
  onFocus: action('onFocus'),
  onMouseDown: action('onMouseDown'),
  onMouseEnter: action('onMouseEnter'),
  onMouseLeave: action('onMouseLeave'),
  onMouseUp: action('onMouseUp'),
  className: 'some-class',
};

storiesOf('ModalWrapper', module)
  .addWithInfo(
    'transactional modal',
    `
      Transactional modals are used to validate user decisions or to gain secondary confirmation from the user.
    `,
    () => (
      <ModalWrapper
        modalProps={modalProps}
        id="transactional-modal"
        buttonTriggerText="Transactional Modal"
        modalLabel="Label (optional)"
        modalHeading="Transactional Modal"
        primaryButtonText="Save"
        secondaryButtonText="Cancel"
        handleSubmit={() => {
          action('onSubmit')();
          return true;
        }}
        shouldCloseAfterSubmit={true}>
        <p className="bx--modal-content__text">
          Transactional modals are used to validate user decisions os to gain
          secondary confirmation from the user. Typically, the modal requests
          either a 'yes' or 'no' response.
        </p>
        <p className="bx--modal-content__text">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
          cursus fermentum risus, sit amet fringilla nunc pellentesque quis.
          Duis quis odio ultrices, cursus lacus ac, posuere felis. Donec
          dignissim libero in augue mattis, a molestie metus vestibulum. Aliquam
          placerat felis ultrices lorem condimentum, nec ullamcorper felis
          porttitor. Nunc at maximus purus. Curabitur sodales eros sit amet
          dolor bibendum gravida. Sed efficitur, arcu imperdiet vestibulum
          ultrices, risus diam ullamcorper arcu, sit amet gravida metus ligula
          quis metus.
        </p>
        <p className="bx--modal-content__text">
          Donec id sapien ex. Duis aliquam tortor nec mollis pulvinar. Fusce sit
          amet libero blandit, sollicitudin est in, tempor lectus. Donec
          convallis condimentum mi eu ultrices. Sed risus ipsum, fermentum ut
          fringilla sed, rutrum eget purus. Morbi molestie dui nisi, consectetur
          posuere ante viverra non. Integer cursus quis risus ut cursus. Aenean
          ut dictum nibh. Cras at leo interdum, ornare elit non, posuere enim.
          Mauris hendrerit nunc eget malesuada congue. Nam velit leo, convallis
          et lobortis ac, semper ut urna. Etiam nec neque urna. Donec sagittis
          eros urna, at aliquet erat consectetur ac. Proin faucibus sed erat
          tempus accumsan. Nam pretium malesuada commodo. Mauris mollis
          elementum neque, sed ornare urna vestibulum non. Curabitur elementum
          in ex in commodo. Donec laoreet consequat sapien. Sed eget tortor
          aliquam, facilisis est vitae, commodo magna. Lorem ipsum dolor sit
          amet, consectetur adipiscing elit. Mauris a lobortis ex. Nulla mollis
          tincidunt sodales.
        </p>
      </ModalWrapper>
    )
  )
  .addWithInfo(
    'passive modal',
    `
      Passive modal notifications should only appear if there is an action the user needs to address immediately.
      Passive modal notifications are persistent on screen.
    `,
    () => (
      <ModalWrapper
        id="passive modal"
        buttonTriggerText="Passive Modal"
        modalLabel="Label (optional)"
        modalHeading="Passive Modal"
        passiveModal>
        <p className="bx--modal-content__text">
          Passive modal notifications should only appear if there is an action
          the user needs to address immediately. Passive modal notifications are
          persistent on screen.
        </p>
      </ModalWrapper>
    )
  )
  .addWithInfo(
    'input modal',
    `
      Input modals are used to follow up with previous user input. These modals should include areas
      for input that the user can interact with, such as forms, dropdowns, selectors, and links. The example
      below shows a Modal Wrapper component with various input components.
    `,
    () => (
      <ModalWrapper
        id="input-modal"
        buttonTriggerText="Input Modal"
        modalHeading="Modal with Inputs"
        handleSubmit={action('onSubmit')}>
        <TextInput
          id="test2"
          placeholder="Hint text here"
          label="Text Input:"
        />
        <br />
        <Select id="select-1" labelText="Select">
          <SelectItem
            disabled
            hidden
            value="placeholder-item"
            text="Pick an option"
          />
          <SelectItem value="option-1" text="Option 1" />
          <SelectItem value="option-2" text="Option 2" />
          <SelectItem value="option-3" text="Option 3" />
        </Select>
        <br />
        <RadioButtonGroup
          name="radio-button-group"
          defaultSelected="default-selected">
          <RadioButton
            value="default-selected"
            id="radio-1"
            labelText="Radio Button label 1"
            className="some-class"
          />
          <RadioButton
            value="standard"
            labelText="Radio Button label 2"
            id="radio-2"
            className="some-class"
          />
          <RadioButton
            value="standard"
            labelText="Radio Button label 3"
            id="radio-3"
            className="some-class"
            disabled
          />
        </RadioButtonGroup>
      </ModalWrapper>
    )
  );
