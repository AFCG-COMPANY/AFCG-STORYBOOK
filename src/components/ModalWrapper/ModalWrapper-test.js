import React from 'react';
import { mount } from 'enzyme';
import ModalWrapper from '../ModalWrapper';

describe('ModalWrapper', () => {
  let mockProps = {};

  beforeEach(() => {
    mockProps = {
      id: 'modal',
      buttonTriggerText: 'Test Modal',
      buttonTriggerClassName: 'btn-trigger',
      modalHeading: 'Transactional Modal',
      modalLabel: 'Test Modal Label',
      primaryButtonText: 'Save',
      secondaryButtonText: 'Cancel',
      handleSubmit: jest.fn(() => true),
      shouldCloseAfterSubmit: true,
    };
  });

  it('should render', () => {
    const wrapper = mount(
      <ModalWrapper {...mockProps}>
        <p className="bx--modal-content__text">Text</p>
      </ModalWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should close after a successful submit action', () => {
    const wrapper = mount(
      <ModalWrapper {...mockProps}>
        <p className="bx--modal-content__text">Text</p>
      </ModalWrapper>
    );
    wrapper.find({ children: mockProps.buttonTriggerText }).simulate('click');
    expect(wrapper.state('isOpen')).toBe(true);

    wrapper.find({ children: mockProps.primaryButtonText }).simulate('click');
    expect(wrapper.state('isOpen')).toBe(false);
  });

  it('should not close after an unsuccessful submit action', () => {
    mockProps.handleSubmit = jest.fn(() => false);
    const wrapper = mount(
      <ModalWrapper {...mockProps}>
        <p className="bx--modal-content__text">Text</p>
      </ModalWrapper>
    );
    wrapper.find({ children: mockProps.buttonTriggerText }).simulate('click');
    expect(wrapper.state('isOpen')).toBe(true);

    wrapper.find({ children: mockProps.primaryButtonText }).simulate('click');
    expect(wrapper.state('isOpen')).toBe(true);
  });
});
