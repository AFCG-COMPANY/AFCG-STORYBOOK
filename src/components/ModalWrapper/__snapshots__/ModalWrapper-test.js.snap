// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper should render 1`] = `
<ModalWrapper
  buttonTriggerClassName="btn-trigger"
  buttonTriggerText="Test Modal"
  disabled={false}
  handleSubmit={[MockFunction]}
  id="modal"
  modalHeading="Transactional Modal"
  modalLabel="Test Modal Label"
  onKeyDown={[Function]}
  primaryButtonText="Save"
  secondaryButtonText="Cancel"
  shouldCloseAfterSubmit={true}
  triggerButtonKind="primary"
>
  <div
    onKeyDown={[Function]}
    role="presentation"
  >
    <Button
      className="btn-trigger"
      disabled={false}
      iconDescription="Provide icon description if icon is used"
      kind="primary"
      onClick={[Function]}
      small={false}
      tabIndex={0}
      type="button"
    >
      <button
        className="btn-trigger bx--btn bx--btn--primary"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        Test Modal
      </button>
    </Button>
    <Modal
      iconDescription="close the modal"
      id="modal"
      modalHeading="Transactional Modal"
      modalLabel="Test Modal Label"
      onKeyDown={[Function]}
      onRequestClose={[Function]}
      onRequestSubmit={[Function]}
      open={false}
      passiveModal={false}
      primaryButtonDisabled={false}
      primaryButtonText="Save"
      secondaryButtonText="Cancel"
    >
      <div
        className="bx--modal bx--modal-tall"
        id="modal"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        tabIndex={-1}
      >
        <div
          className="bx--modal-container"
          role="dialog"
        >
          <div
            className="bx--modal-header"
          >
            <h4
              className="bx--modal-header__label"
            >
              Test Modal Label
            </h4>
            <h2
              className="bx--modal-header__heading"
            >
              Transactional Modal
            </h2>
            <button
              className="bx--modal-close"
              onClick={[Function]}
              type="button"
            >
              <Icon
                className="bx--modal-close__icon"
                description="close the modal"
                fillRule="evenodd"
                name="close"
                role="img"
              >
                <svg
                  alt="close the modal"
                  aria-label="close the modal"
                  className="bx--modal-close__icon"
                  fillRule="evenodd"
                  height="10"
                  name="close"
                  role="img"
                  viewBox="0 0 10 10"
                  width="10"
                >
                  <title>
                    close the modal
                  </title>
                  <path
                    d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
                    key="key0"
                  />
                </svg>
              </Icon>
            </button>
          </div>
          <div
            className="bx--modal-content"
          >
            <p
              className="bx--modal-content__text"
            >
              Text
            </p>
          </div>
          <div
            className="bx--modal-footer"
          >
            <div
              className="bx--modal__buttons-container"
            >
              <Button
                disabled={false}
                iconDescription="Provide icon description if icon is used"
                kind="secondary"
                onClick={[Function]}
                small={false}
                tabIndex={0}
                type="button"
              >
                <button
                  className="bx--btn bx--btn--secondary"
                  disabled={false}
                  onClick={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  Cancel
                </button>
              </Button>
              <Button
                disabled={false}
                iconDescription="Provide icon description if icon is used"
                inputref={
                  Object {
                    "current": <button
                      class="bx--btn bx--btn--primary"
                      inputref="[object Object]"
                      tabindex="0"
                      type="button"
                    >
                      Save
                    </button>,
                  }
                }
                kind="primary"
                onClick={[Function]}
                small={false}
                tabIndex={0}
                type="button"
              >
                <button
                  className="bx--btn bx--btn--primary"
                  disabled={false}
                  inputref={
                    Object {
                      "current": <button
                        class="bx--btn bx--btn--primary"
                        inputref="[object Object]"
                        tabindex="0"
                        type="button"
                      >
                        Save
                      </button>,
                    }
                  }
                  onClick={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  Save
                </button>
              </Button>
            </div>
          </div>
        </div>
      </div>
    </Modal>
  </div>
</ModalWrapper>
`;
